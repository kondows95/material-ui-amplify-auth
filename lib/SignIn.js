"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var core_1 = require("@material-ui/core");
var react_intl_1 = require("react-intl");
var FormTitle_1 = __importDefault(require("./FormTitle"));
var MyLink_1 = __importDefault(require("./MyLink"));
var material_ui_basic_parts_1 = require("material-ui-basic-parts");
var Password_1 = __importDefault(require("./Password"));
var SignIn = function (props) {
    var formatMessage = react_intl_1.useIntl().formatMessage;
    var _a = react_1.default.useState({ email: '', password: '' }), form = _a[0], setForm = _a[1];
    var handleChangeValue = function (fieldName) { return function (event) {
        var newForm = __assign({}, form);
        newForm[fieldName] = event.currentTarget.value;
        setForm(newForm);
    }; };
    var handleSubmit = function (event) {
        event.preventDefault();
        props.signIn(form['email'], form['password']);
    };
    var handleForgotPassword = function (event) {
        event.preventDefault();
        props.changeAuthState('forgotPassword');
    };
    var handleSignUp = function (event) {
        event.preventDefault();
        props.changeAuthState('signUp');
    };
    var msgId = 'MUAA.SignIn';
    var msgIdTitle = msgId + '.Title';
    var msgIdEmail = msgId + '.Email';
    var msgIdSubmit = msgId + '.Submit';
    var msgIdPassword = msgId + '.Password';
    var msgIdForgotPassword = msgId + '.ForgotPassword';
    var msgIdSignUp = msgId + '.SignUp';
    var link = (react_1.default.createElement(core_1.Box, { mb: 1, display: "flex", justifyContent: "center", alignItems: "center" },
        react_1.default.createElement(MyLink_1.default, { label: formatMessage({ id: msgIdForgotPassword, defaultMessage: 'Forgot password?' }), "data-testid": msgIdForgotPassword, onClick: handleForgotPassword })));
    if (props.enableSignUp) {
        link = (react_1.default.createElement(core_1.Grid, { container: true },
            react_1.default.createElement(core_1.Grid, { item: true, xs: 12 }, link),
            react_1.default.createElement(core_1.Grid, { item: true, xs: 12 },
                react_1.default.createElement(core_1.Box, { mb: 1, display: "flex", justifyContent: "center", alignItems: "center" },
                    react_1.default.createElement(MyLink_1.default, { label: formatMessage({
                            id: msgIdSignUp,
                            defaultMessage: 'No account? Create account',
                        }), "data-testid": msgIdSignUp, onClick: handleSignUp })))));
    }
    var content = (react_1.default.createElement(core_1.Container, { component: "main", maxWidth: "xs" },
        react_1.default.createElement("form", { onSubmit: handleSubmit },
            react_1.default.createElement(core_1.Box, { display: "flex", flexDirection: "column", mt: 8 },
                react_1.default.createElement(FormTitle_1.default, { "data-testid": msgIdTitle, label: formatMessage({ id: msgIdTitle, defaultMessage: 'Sign In' }) }),
                react_1.default.createElement(core_1.Box, { display: "flex", justifyContent: "center", fontWeight: 600, color: "error.main" }, props.error),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(core_1.TextField, { autoComplete: "email", type: "email", onChange: handleChangeValue('email'), value: form.email, label: formatMessage({ id: msgIdEmail, defaultMessage: 'Email Address' }), variant: "outlined", required: true, fullWidth: true, inputProps: { 'data-testid': msgIdEmail } })),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(Password_1.default, { label: formatMessage({ id: msgIdPassword, defaultMessage: 'Password' }), testId: msgIdPassword, value: form.password, onChange: handleChangeValue('password') })),
                react_1.default.createElement(core_1.Box, { width: "100%", mt: 4, mb: 2, className: "relative" },
                    react_1.default.createElement(material_ui_basic_parts_1.MyButton, { type: "submit", label: formatMessage({ id: msgIdSubmit, defaultMessage: 'Sign In' }), color: "primary", loading: props.loading, "data-testid": msgIdSubmit })),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 }, link)))));
    return props.authState === 'signIn' ? content : null;
};
exports.default = SignIn;
