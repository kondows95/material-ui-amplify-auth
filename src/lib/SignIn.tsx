import React, { MouseEvent, FormEvent, ChangeEvent } from 'react';
import { Box, Container, TextField, Grid } from '@material-ui/core';
import { useIntl } from 'react-intl';
import FormTitle from './FormTitle';
import MyLink from './MyLink';
import { MyButton } from 'material-ui-basic-parts';
import Password from './Password';

type Props = {
    authState: string;
    changeAuthState: (state: string) => void;
    signIn: (email: string, password: string) => void;
    loading: boolean;
    error: string;
    enableSignUp?: boolean;
};

const SignIn: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();
    type Form = { [key: string]: string };
    const [form, setForm] = React.useState<Form>({ email: '', password: '' });

    type FormKey = 'email' | 'password';
    const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>): void => {
        const newForm = { ...form };
        newForm[fieldName] = event.currentTarget.value;
        setForm(newForm);
    };

    const handleSubmit = (event: FormEvent): void => {
        event.preventDefault();
        props.signIn(form['email'], form['password']);
    };

    const handleForgotPassword = (event: MouseEvent): void => {
        event.preventDefault();
        props.changeAuthState('forgotPassword');
    };

    const handleSignUp = (event: MouseEvent): void => {
        event.preventDefault();
        props.changeAuthState('signUp');
    };

    const msgId = 'MUAA.SignIn';
    const msgIdTitle = msgId + '.Title';
    const msgIdEmail = msgId + '.Email';
    const msgIdSubmit = msgId + '.Submit';
    const msgIdPassword = msgId + '.Password';
    const msgIdForgotPassword = msgId + '.ForgotPassword';
    const msgIdSignUp = msgId + '.SignUp';

    let link = (
        <Box mb={1} display="flex" justifyContent="center" alignItems="center">
            <MyLink
                label={formatMessage({ id: msgIdForgotPassword, defaultMessage: 'Forgot password?' })}
                data-testid={msgIdForgotPassword}
                onClick={handleForgotPassword}
            />
        </Box>
    );
    if (props.enableSignUp) {
        link = (
            <Grid container>
                <Grid item xs={12}>
                    {link}
                </Grid>
                <Grid item xs={12}>
                    <Box mb={1} display="flex" justifyContent="center" alignItems="center">
                        <MyLink
                            label={formatMessage({
                                id: msgIdSignUp,
                                defaultMessage: 'No account? Create account',
                            })}
                            data-testid={msgIdSignUp}
                            onClick={handleSignUp}
                        />
                    </Box>
                </Grid>
            </Grid>
        );
    }

    const content = (
        <Container component="main" maxWidth="xs">
            <form onSubmit={handleSubmit}>
                <Box display="flex" flexDirection="column" mt={8}>
                    <FormTitle
                        data-testid={msgIdTitle}
                        label={formatMessage({ id: msgIdTitle, defaultMessage: 'Sign In' })}
                    />
                    <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
                        {props.error}
                    </Box>
                    <Box width="100%" my={2}>
                        <TextField
                            autoComplete="email"
                            type="email"
                            onChange={handleChangeValue('email')}
                            value={form.email}
                            label={formatMessage({ id: msgIdEmail, defaultMessage: 'Email Address' })}
                            variant="outlined"
                            required
                            fullWidth
                            inputProps={{ 'data-testid': msgIdEmail }}
                        />
                    </Box>
                    <Box width="100%" my={2}>
                        <Password
                            label={formatMessage({ id: msgIdPassword, defaultMessage: 'Password' })}
                            testId={msgIdPassword}
                            value={form.password}
                            onChange={handleChangeValue('password')}
                        />
                    </Box>
                    <Box width="100%" mt={4} mb={2} className="relative">
                        <MyButton
                            type="submit"
                            label={formatMessage({ id: msgIdSubmit, defaultMessage: 'Sign In' })}
                            color="primary"
                            loading={props.loading}
                            data-testid={msgIdSubmit}
                        />
                    </Box>
                    <Box width="100%" my={2}>
                        {link}
                    </Box>
                </Box>
            </form>
        </Container>
    );
    return props.authState === 'signIn' ? content : null;
};

export default SignIn;
