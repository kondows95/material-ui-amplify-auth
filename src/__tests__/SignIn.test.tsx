import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import SignIn from '../lib/SignIn';

let tLib: TestLib;
let changeAuthState: jest.Mock;
let signIn: jest.Mock;

beforeEach(() => {
    changeAuthState = jest.fn();
    signIn = jest.fn();
});

describe('Basic Usage', () => {
    const msgs = {
        'MUAA.SignIn.Title': '#MUAA.SignIn.Title',
        'MUAA.SignIn.Email': '#MUAA.SignIn.Email',
        'MUAA.SignIn.Password': '#MUAA.SignIn.Password',
        'MUAA.SignIn.Submit': '#MUAA.SignIn.Submit',
        'MUAA.SignIn.ForgotPassword': '#MUAA.SignIn.ForgotPassword',
        'MUAA.SignIn.SignUp': '#MUAA.SignIn.SignUp',
    };
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={msgs}>
                    <SignIn
                        authState={'signIn'}
                        changeAuthState={changeAuthState}
                        signIn={signIn}
                        loading={false}
                        enableSignUp={true}
                        error={'MyError'}
                    />
                </IntlProvider>
            )
        );
    });

    it('If ForgotPassword clicked, authState will be change.', () => {
        tLib.click('MUAA.SignIn.ForgotPassword');
        expect(changeAuthState).toHaveBeenCalledWith('forgotPassword');
    });

    it('If SignUp clicked, authState will be change.', () => {
        tLib.click('MUAA.SignIn.SignUp');
        expect(changeAuthState).toHaveBeenCalledWith('signUp');
    });

    it('Click submit button', () => {
        tLib.changeValue('MUAA.SignIn.Email', 'test@example.com');
        tLib.changeValue('MUAA.SignIn.Password', 'abc123');
        tLib.click('MUAA.SignIn.Submit');
        expect(signIn).toHaveBeenCalledWith('test@example.com', 'abc123');
    });

    it('You can set label by Intl', () => {
        const html = tLib.render.container.innerHTML;
        expect(html).toContain('#MUAA.SignIn.Title');
        expect(html).toContain('#MUAA.SignIn.Email');
        expect(html).toContain('#MUAA.SignIn.Password');
        expect(html).toContain('#MUAA.SignIn.Submit');
        expect(html).toContain('#MUAA.SignIn.ForgotPassword');
    });

    it('You can show password.', () => {
        expect(tLib.get('MUAA.SignIn.Password').getAttribute('type')).toBe('password');
        tLib.click('MUAA.SignIn.Password.Visible');
        expect(tLib.get('MUAA.SignIn.Password').getAttribute('type')).toBe('text');
    });

    it('You can set error message', () => {
        expect(tLib.render.container.innerHTML).toContain('MyError');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

it('Default Intl labels', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <SignIn
                    authState={'signIn'}
                    changeAuthState={changeAuthState}
                    signIn={signIn}
                    loading={true}
                    enableSignUp={true}
                    error={''}
                />
            </IntlProvider>
        )
    );
    const html = tLib.render.container.innerHTML;
    expect(html).toContain('Sign In');
    expect(html).toContain('Email Address');
    expect(html).toContain('Password');
    expect(html).toContain('No account? Create account');
    expect(html).toContain('Forgot password?');
});

it('You can hide SignUp.', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <SignIn
                    authState={'signIn'}
                    changeAuthState={changeAuthState}
                    signIn={signIn}
                    loading={true}
                    error={''}
                />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.innerHTML).not.toContain('No account? Create account');
});

it('If authState is not signIn, return empty.', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <SignIn
                    authState={'forgotPassword'}
                    changeAuthState={changeAuthState}
                    signIn={signIn}
                    loading={true}
                    enableSignUp={true}
                    error={''}
                />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.textContent).toBe('');
});
