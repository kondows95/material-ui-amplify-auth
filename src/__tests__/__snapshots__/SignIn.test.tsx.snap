// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <form>
      <div
        class="MuiBox-root MuiBox-root-935"
      >
        <div
          class="MuiBox-root MuiBox-root-936"
          data-testid="MUAA.SignIn.Title"
        >
          <div
            class="MuiBox-root MuiBox-root-937"
          >
            <div
              class="MuiBox-root MuiBox-root-938"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-948"
          >
            #MUAA.SignIn.Title
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-949"
        >
          MyError
        </div>
        <div
          class="MuiBox-root MuiBox-root-950"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              #MUAA.SignIn.Email
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="MUAA.SignIn.Email"
                required=""
                type="email"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1009 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-1011"
                >
                  <span>
                    #MUAA.SignIn.Email *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-1013"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="standard-adornment-password"
            >
              #MUAA.SignIn.Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              required=""
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                data-testid="MUAA.SignIn.Password"
                id="standard-adornment-password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                  data-testid="MUAA.SignIn.Password.Visible"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1009 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-1010"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-1033 relative"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="MUAA.SignIn.Submit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              #MUAA.SignIn.Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-1064"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            data-testid="MUAA.SignIn.ForgotPassword"
            href="#"
          >
            #MUAA.SignIn.ForgotPassword
          </a>
        </div>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
